File: app/src/main/java/com/hgianastasio/biblioulbrav2/views/adapters/RenewLoanBookAdapter.java
package com.hgianastasio.biblioulbrav2.views.adapters;

import android.content.Context;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.hgianastasio.biblioulbrav2.R;
import com.hgianastasio.biblioulbrav2.models.loanbooks.LoanBookModel;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by heitor_12 on 04/05/17.
 */

public class RenewLoanBookAdapter extends RecyclerView.Adapter<RenewLoanBookAdapter.ViewHolder> {
    private List<LoanBookModel> list;
    private Context mContext;

    public RenewLoanBookAdapter(List<LoanBookModel> list, Context mContext) {
        this.list = list;
        this.mContext = mContext;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.renew_book_item,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        LoanBookModel model = list.get(position);
        holder.tvBookTitle.setText(model.getTitle());
        holder.tvDeadline.setText(model.getDeadline().concat(model.isSingleCopy()?"*":""));
        int color;
        if(model.isOverdue())
            color = R.color.redPenalty;
        else if(model.wasJustRenewed())
            color = R.color.yellowWarning;
        else
            color = R.color.greenPenalty;

        holder.tvDeadline.setTextColor(ContextCompat.getColor(mContext,color));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }


    class ViewHolder extends RecyclerView.ViewHolder{
        @BindView(R.id.tvBookTitle) TextView tvBookTitle;
        @BindView(R.id.tvDeadline) TextView tvDeadline;

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this,itemView);
        }
    }
}

File: app/src/main/java/com/hgianastasio/biblioulbrav2/views/adapters/RenewLoanBookAdapter.kt
package com.hgianastasio.biblioulbrav2.views.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.hgianastasio.biblioulbrav2.R
import com.hgianastasio.biblioulbrav2.databinding.RenewBookItemBinding
import com.hgianastasio.biblioulbrav2.models.loanbooks.LoanBookModel

/**
 * Created by heitor_12 on 04/05/17.
 */
class RenewLoanBookAdapter(private val list: List<LoanBookModel?>, private val mContext: Context?) : RecyclerView.Adapter<RenewLoanBookAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(mContext).inflate(R.layout.renew_book_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val model = list[position]
        holder.binding.tvBookTitle.text = model?.title
        holder.binding.tvDeadline.text = model?.deadline + if (model!!.isSingleCopy) "*" else ""
        val color: Int
        color = if (model.isOverdue) R.color.redPenalty else if (model.wasJustRenewed()) R.color.yellowWarning else R.color.greenPenalty
        holder.binding.tvDeadline.setTextColor(ContextCompat.getColor(mContext!!, color))
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val binding = RenewBookItemBinding.bind(itemView)
    }

}